File: ruby/tutorial-horizon/SumCloseToZero.rb
a =  [1, 60, -10, 70, -80, 85]

#a =  [-5,-5,-8];


sum = a.first + a.last
a.each_with_index do |index, val|
    a.each_with_index do |index2, val2|
       next if  index == index2
       temp = val + val2
       if  sum.abs > temp.abs
         sum = temp
       end
     end
end
 p sum

File: ruby/tutorial-horizon/sumCloseToZero.php
<?php

$array = [-5,-5,-8];
$min_sum = $array[0] + $array[1];
for( $l = 0; $l <= count($array) -1;  $l++ ){
    for( $j = $l + 1; $j <= count($array) -1;  $j++ ){
        $temp =  $array[$l] + $array[$j];
        if (abs($temp) < abs($min_sum)) 
            $min_sum = $temp;
    }
}

echo $min_sum;